src/plugins/wizard/public/visualizations/metric/to_expression.ts

```diff
@@ -91,13 +91,18 @@ interface MetricRootState extends RootState {
   style: MetricOptionsDefaults;
 }
 
-export const toExpression = async ({ style: styleState, visualization }: MetricRootState) => {
+export const toExpression = async (
+  { style: styleState, visualization }: MetricRootState,
+  indexPatterns,
+  aggs
```

Comment:
do we need these? It looks like all you need them for is if you want to manually pass in the index pattern service or the agg service. Both of which dont come from the `wizard` plugin. Since we intend to have visualization types that aren't a part of the `wizard` plugin, we dont need to pass this in from the wizard plugin. Especially since both the services we use are unmodified in any way :)

Reply:
Yeah, the point is that we need access to these services to calculate the expression. In the d&d app/canvas, you can get these from the wizard start services; but from an embeddable context, you bring your own start services.

Reply:
fixed/reverted to getters/setters in #1947